tests/unit/auth/test_sigv4.py

```diff
@@ -507,6 +507,11 @@ def test_canonical_request(self):
 
     def test_non_string_headers(self):
         self.awesome_bucket_request.headers['Content-Length'] = 8
+        # Headers in canonical order are alphabetized by key alone.
+        # This test ensures we are not alphabetizing based on the header/value
+        # separator as well:
```

Comment:
"because ':' has a lower ordinal value than '-'"?

Reply:
exactly! That's what the bug was :)